### PR TITLE
feat: support eslint v8

### LINKED ISSUE #192: eslint 8 compatibility
Hi guys!

Please tell us if `eslint-config-standard` is compatible with the new version of ESlint? 
and if yes please add it to peerDependencies

### PR DESCRIPTION
**Pull request by Divlo**
ESLint v8 has been released stable. :tada:
It's time to support `eslint@8`.

See: https://eslint.org/blog/2021/10/eslint-v8.0.0-released

Blocked by:
- [x] https://github.com/import-js/eslint-plugin-import/issues/2211
- [x] https://github.com/mysticatea/eslint-plugin-node/pull/295 (replaced by  https://www.npmjs.com/package/eslint-plugin-n)
- [x] https://github.com/xjamundx/eslint-plugin-promise/issues/218

Closes #192

### CODE REVIEW

**Comment by voxpelli** [comment]
Should we consider adding any of the news `eslint:recommended` rules? Probably wait for a major after this one, right?

* no-loss-of-precision
* no-nonoctal-decimal-escape
* no-unsafe-optional-chaining
* no-useless-backreference

**Comment by Divlo** [comment]
> @Divlo Do you know if `eslint-plugin-node` _requires_ an update? As its set to be compatible with all future versions of ESLint. Is ESLint 8 breaking it in any way?

I don't know if there's something breaking with ESLint v8.
We should probably wait a reply to your comment, to see how it goes: https://github.com/mysticatea/eslint-plugin-node/issues/294#issuecomment-943688252 @voxpelli

**Comment by Divlo** [comment]
> Should we consider adding any of the news `eslint:recommended` rules? Probably wait for a major after this one, right?
> 
> * no-loss-of-precision
> * no-nonoctal-decimal-escape
> * no-unsafe-optional-chaining
> * no-useless-backreference

Good idea! :+1:
I think we can alreay include these new rules in the next major version as this PR will be released in a major version anyway.

But maybe just to make this PR small and separated, we could include these rules in another PR.

File: .eslintrc.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "parserOptions": {
-    "ecmaVersion": 2021,
+    "ecmaVersion": "latest",
```
[the remark this answers is not in the document]
Comment (Divlo) [reply .eslintrc.json]: My mistake, it doesn't exist yet. I reverted it back to `2021`.

**Comment by Divlo** [comment]
CI is failing, because `eslint-plugin-promise` set explicitly the `peerDependencies` : https://github.com/xjamundx/eslint-plugin-promise/blob/1f982da02579fcc18a13880b4c4904e530d3a54b/package.json#L41.

Last commit, we used an older version of `npm` that didn't fail with unmet `peerDependencies` but only shows a warning:

`npm WARN eslint-plugin-promise@5.1.0 requires a peer of eslint@^7.0.0 but none is installed. You must install peer dependencies yourself.`.

**Comment by voxpelli** [comment]
There is a eslint 8 fork of eslint-plugin-node now which we can try out: https://www.npmjs.com/package/eslint-plugin-n

**Comment by aladdin-add** [comment]
you can simply fix the ci by replace `node/foo` => `n/foo` in the config.

I should have updated the docs in the fork. it's still in my todo list :)

**Comment by Divlo** [comment]
> you can simply fix the ci by replace `node/foo` => `n/foo` in the config.
> 
> I should have updated the docs in the fork. it's still in my todo list :)

Thanks, it indeed fixes the CI. :smile:

**Review by aladdin-add — APPROVED** [review]

**Review by midudev — APPROVED** [review]

**Comment by Nagarian** [comment]
It seems that eslint-plugin-promise as been updated to eslint 8 https://github.com/xjamundx/eslint-plugin-promise/blob/development/CHANGELOG.md

**Comment by J3m5** [comment]
> It seems that eslint-plugin-promise as been updated to eslint 8 https://github.com/xjamundx/eslint-plugin-promise/blob/development/CHANGELOG.md

Related issue: https://github.com/standard/eslint-config-standard/issues/200

**Comment by fmal** [comment]
anything blocking this being merged now?

**Comment by Divlo** [comment]
> anything blocking this being merged now?

Nothing is blocking, it needs reviews! :smile:

**Review by skjnldsv — APPROVED** [review]

**Review by markusl — APPROVED** [review]

**Comment by wesleytodd** [comment]
Not that my approval means much as a non-contributor, but I pointed my project at this branch and ran with `eslint@8` and it worked great.  Really excited to finally be able to use private class properties with `standard` 🥇! Thanks for the good work team!

**Review by wesleytodd — APPROVED** [review]

File: .eslintrc.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (aladdin-add) [reply .eslintrc.json]: IMHO, it needs to be 2022 - to use class fileds, TLA...

**Review by innocenzi — APPROVED** [review]

**Review by MartinX3 — APPROVED** [review]

**Comment by yoyo837** [comment]
Friendly ping... @Divlo

**Comment by voxpelli** [comment]
It’s not Divlo but the rest of us maintainers that are a bit slow at reviewing this, I myself have been taking time of programming now during the Christmas holiday, but hope to get around to this one any day now

**Comment by danger89** [comment]
Yes, finally please.

**Review by voxpelli — APPROVED** [review]

**Comment by voxpelli** [comment]
I have now rolled a `17.0.0-0` _prerelease_  of this.

You can install it using `eslint-config-standard@next`.

Being a pre-release, this one comes with no guarantees in regards to support / breaking changes etc. It exists to facilitate early feedback, from the wider community as well from our selves.

Next step now is to ship a prerelease of `standard` itself, to further validate the changes introduced here and minimize the risk for unintended breakage.

**Review by posva — COMMENTED** [review]
~~Thanks!~~ oops, wrong repo

**Pull request merged**